vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/SortingIT.java

```diff
@@ -61,6 +62,141 @@ public void setInitialSortOrder_changeOrderFromServer_sortIndicatorsUpdated() {
         findElement(By.id("sort-by-age")).click();
         assertAscendingSorter("Age");
     }
+    @Test
+    public void keepSortStatesAfterReAttach() {
+        WebElement btnAttach = findElement(By.id("btn-attach"));
+        WebElement btnRemove = findElement(By.id("btn-deattach"));
+
+        // Attach grid
+        btnAttach.click();
+
+        WebElement secondGrid = findElement(By.id("second-grid"));
+        GridElement secondGridElement = $(GridElement.class).get(1);
+        // Assert sort from server side before reattach
+        Assert.assertEquals("asc", secondGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+
+        // Deattach
+        btnRemove.click();
+        // Reattach
+        btnAttach.click();
+
+        secondGrid = findElement(By.id("second-grid"));
+        Assert.assertEquals("asc", secondGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+
+        // Assert sort reattach without deattach and sort from server
+        // Continue reattach
+        secondGridElement = $(GridElement.class).get(1);
+        String textFirstStringColumnBeforeReattch = secondGridElement.getCell(0, 1).getText();
+        String textFirstIndexColumnBeforeReattch = secondGridElement.getCell(0, 2).getText();
+        String textFirstRandomColumnBeforeReattch = secondGridElement.getCell(0, 3).getText();
+
+        btnAttach.click();
+
+        secondGridElement = $(GridElement.class).get(1);
+        secondGrid = findElement(By.id("second-grid"));
+        String textFirstStringColumnAfterReattch = secondGridElement.getCell(0, 1).getText();
+        String textFirstIndexColumnAfterReattch = secondGridElement.getCell(0, 2).getText();
+        String textFirstRandomColumnAfterReattch = secondGridElement.getCell(0, 3).getText();
+        Assert.assertEquals("asc", secondGrid.findElements(By.tagName("vaadin-grid-sorter"))
+                .get(1).getAttribute("direction"));
+        Assert.assertEquals(textFirstStringColumnBeforeReattch, textFirstStringColumnAfterReattch);
+        Assert.assertEquals(textFirstIndexColumnBeforeReattch, textFirstIndexColumnAfterReattch);
+        Assert.assertEquals(textFirstRandomColumnBeforeReattch, textFirstRandomColumnAfterReattch);
+
+        // Assert sort with rather than 1 column, handle from client
```

Comment:
This is a bit hard to understand. Can you rephrase?

Reply:
it means that I click and sort column, from client, not using sort(abc) from java side. 
And sort with 2-3-4 columns, instead only one column.

Reply:
but it is an old test, my new test to test the flow: attach - detach and attach again, and check the sort states are kept or not.